src/main/java/org/jabref/logic/importer/fetcher/CrossRef.java

```diff
@@ -24,96 +37,145 @@
  *
  * See https://github.com/CrossRef/rest-api-doc
  */
-public class CrossRef {
-    private static final Log LOGGER = LogFactory.getLog(CrossRef.class);
+public class CrossRef implements IdParserFetcher<DOI>, EntryBasedParserFetcher, SearchBasedParserFetcher, IdBasedParserFetcher {
```

Comment:
To be honest, the hierarchy of interfaces etc. has gotten way to complicated for the fetchers. I think its really hard to get an idea whats happening right now. With this PR it even got harder to keep track.